src/devices/bus/nes/bootleg.cpp

```diff
@@ -944,6 +961,43 @@ void nes_smb3p_device::write_h(offs_t offset, uint8_t data)
 	}
 }
 
+/*-------------------------------------------------
+
+ BTL-CONTRAJ
+
+ Games: Contra (J) pirate
+
+ NES 2.0: mapper 326
+
+ In MAME: Supported.
+
+ TODO: Find out why this crashes in the intro story.
+ Bootleggers missed the bug? Bad dump? This differs
+ only in a couple hundred bytes from contraj, so there
+ aren't that many places for the problem to hide.
+
+ -------------------------------------------------*/
+
+void nes_btl_cj_device::write_h(offs_t offset, u8 data)
+{
+	LOG_MMC(("btl_cj write_h, offset: %04x, data: %02x\n", offset, data));
+
+	if (BIT(offset, 4))
+	{
+		offset &= 0x0f;
+		if (offset < 0x08)
+			chr1_x(offset, data, CHRROM);
+		else
+			set_nt_page(offset & 0x03, CIRAM, data & 1, 1);
```

Comment:
Wouldn’t this be more clearly expressed like:
```
		if (BIT(offset, 3))
			set_nt_page(offset & 0x03, CIRAM, data & 1, 1);
		else
			chr1_x(offset & 0x03, data, CHRROM);
```

Neither branch actually uses bit 3 of the offset for the bank number.

Reply:
Updated. Yes, that's a bit better. You got me though! I got garbled graphics on the first try because the second mask is 7 :smile: